server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/processor/CICSTranslateMandatorySectionProcess.java

```diff
@@ -46,10 +47,12 @@ private void insertLinkageSection(ProgramNode programNode) {
     if (dataDivisionNode.isPresent()) {
       addLinkageNode(dataDivisionNode.get());
     } else {
+      Locality locality = Locality.builder().uri(programNode.getLocality().getUri()).build();
```

Comment:
I don't understand these changes.

Reply:
It's for generated variables. We now expect them inside the program (before the END) and the virtual division node shouldn't overlap whole program, so we can still search nodes inside it.

Reply:
Should there be a range associated with the node (even empty but generally non-zero)?

Reply:
It should https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/9d091f48ebd47630904d8fbde0a6a494add5c104/server/common/src/main/java/org/eclipse/lsp/cobol/common/model/Locality.java#L43

Reply:
but that's always [[0,0], [0,0]] won't that cause any issues? Should it point e.g. to the beginning of the program instead?

Reply:
Should be fine, but ill put the program start there just in case.